### PR TITLE
Fix calendar previous month bug

### PR DESCRIPTION
**Pull request by kyle-annen**
* Fix previous month button bug (array index out of range)

### CODE REVIEW

File: app/views/events/index.html.erb [hunk 1]
```diff
@@ -2,32 +2,33 @@
...
+  <div class="hidden-sm-down col-md-3 calendar-day-label">Tues</div> 
+  <div class="hidden-sm-down col-md-3 calendar-day-label">Wed</div>
+  <div class="hidden-sm-down col-md-3 calendar-day-label">Thurs</div>
+</div>
+<br/>
+<div class="container">
+  <% @events.keys.each do |week_num| %>
     <div class="row">
       <% if @events.keys.min == week_num %>
         <% (4 - @events[week_num].keys.length).times do %>
```
Comment (pwdd) [inline app/views/events/index.html.erb]: Why `4`? What's `4`?

File: app/views/events/index.html.erb [hunk 2]
```diff
@@ -38,32 +39,35 @@
...
-      <div class="col-md-3 calendar-cell">
-        <span class="calendar-date">
-        <%=  Date.commercial(@date.year, week_num, day_num).day %>
-        </span>
+        <div class="col-md-3 calendar-cell-hover">
+          <span class="calendar-date">
+            <%=  Date.commercial(@date.year, week_num, day_num).day %>
+          </span>
         </br>
         <% if @events[week_num][day_num].size > 0 %>
```
Comment (pwdd) [inline app/views/events/index.html.erb]: Is it possible to remove all this logic (lines 47 to 69) from the template and maybe add some tests to make sure the problem was fixed?

File: app/views/events/index.html.erb [hunk 3]
```diff
@@ -2,32 +2,33 @@
...
-    <div class="hidden-sm-down col-md-3 calendar-day-label">Mon</div>
-    <div class="hidden-sm-down col-md-3 calendar-day-label">Tues</div> 
-    <div class="hidden-sm-down col-md-3 calendar-day-label">Wed</div>
-    <div class="hidden-sm-down col-md-3 calendar-day-label">Thurs</div>
-  </div>
-  <br/>
-  <div class="container">
-    <% @events.keys.each do |week_num| %>
+</div>
+<br/>
```
Comment (pwdd) [inline app/views/events/index.html.erb:21]: What are all these `<br>` doing? Are they doing the job of some margin or padding between elements?

File: app/views/events/index.html.erb (hunk 1 above)
Comment (kyle-annen) [reply app/views/events/index.html.erb]: 4 is the week length for the calendar.  This conditional is how the padding on the first week of the month is displayed.  It creates offset columns by calculating the difference between the days in the week subtracted from the days per week displayed.  

I have created an instance variable in the controller to abstract the magic number.

File: app/views/events/index.html.erb (hunk 3 above)
Comment (kyle-annen) [reply app/views/events/index.html.erb:21]: I have removed all the `<br>` tags and have replaced their functionality with proper css.

File: app/views/events/index.html.erb (hunk 2 above)
Comment (kyle-annen) [reply app/views/events/index.html.erb]: I have moved the logic to the helper, written tests for the logic.  

I was wondering if you would suggest writing acceptance tests or if the tests added for the logic would suffice?

**Review by pwdd — APPROVED** [review]

**Comment by kyle-annen** [comment]
Thanks for comments and review Aina!

**Pull request merged**